### PR TITLE
Add ESNext syntax to meta block tutorial

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Adds ESNext syntax example for the Meta Block tutorial
It initially was only written using ES5

## How has this been tested?

Documentation, confirm works as expected.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Nice work.

I applied WordPress code styles to the code examples include in f5e94311a650b3e72d300e90f5debbe368da9a0a given that I develop inner ESLint capabilities in my brain :)

There was also a small mistake in the value prop of `TextField` (which I renamed back the original component name, too).

File: docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md [hunk 1]
```diff
@@ -53,9 +55,52 @@ Add this code to your JavaScript file (this tutorial will call the file `myguten
...
 			return null;
 		}
-	});
-})( window.wp );
+	} );
+} )( window.wp );
 ```
+{% ESNext %}
+```jsx
+const { registerBlockType } = wp.blocks;
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]:
```suggestion
import { registerBlockType } from '@wordpress/blocks';
```

File: docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md [hunk 2]
```diff
@@ -53,9 +55,52 @@ Add this code to your JavaScript file (this tutorial will call the file `myguten
...
+const { TextControl } = wp.components;
```
Comment (nosolosw) [inline docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]: import { TextControl } from '@wordpress/components';

File: docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md [hunk 3]
```diff
@@ -53,9 +55,52 @@ Add this code to your JavaScript file (this tutorial will call the file `myguten
...
 		}
-	});
-})( window.wp );
+	} );
+} )( window.wp );
 ```
+{% ESNext %}
+```jsx
+import { registerBlockType } from '@wordpress/blocks';
+import { TextControl } from '@wordpress/components';
```
Comment (youknowriad) [inline docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]: I'm personally concerned about the change in the imports as it's something that doesn't work unless you define externals. While using globals signals that we should rely on `npm` packages to be bundled here. Think of any JavaScript developper reading this, I'm pretty sure they won't understand that this needs point to globals.

Comment (nosolosw) [reply docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]: That's a fair point I missed. Trying to put together my thoughts on this:

- **in tutorials, handbook, etc. we should promote globals**. These are tailored for WordPress developers, so the assumption is that they're building something to integrate with WordPress. Having both ES5 and ESNext is helpful for onboarding, but promoting globals makes sense here.

- **in package info (READMEs, JSDoc examples, etc) we should promote `import` syntax and avoid the globals**. Packages should be agnostic of the host they're bundled with (WordPress injecting the globals). It's not that important a goal to have good coverage for ES5 here.

This isn't perfect, though. For example: when we expose JSDoc examples in the handbook, they'll contain the `import` syntax. 

Does this sound like a good compromise? I can prepare a PR to fix this.

Comment (nosolosw) [reply docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]: For reference: things will be even more confusing in a few days, once we ship the zero-config `wp-scripts build` command that include externals (and the reason I thought about proposing this) :man_shrugging:

Comment (gziolo) [reply docs/designers-developers/developers/tutorials/metabox/meta-block-3-add.md]: > * **in package info (READMEs, JSDoc examples, etc) we should promote `import` syntax and avoid the globals**. Packages should be agnostic of the host they're bundled with (WordPress injecting the globals). It's not that important a goal to have good coverage for ES5 here.

Agreed on this one. I would be personally fine of using imports only for ESNext examples.

For tutorials and handbook in general, using `wp` globals is a very good idea.

**Pull request merged**